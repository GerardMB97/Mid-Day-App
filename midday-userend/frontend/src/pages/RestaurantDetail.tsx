import React from 'react';
import { Text } from 'react-native';

export default function RestaurantDetail () {
  return <Text>Hello Im restaurantDetailComponent</Text>;
}
