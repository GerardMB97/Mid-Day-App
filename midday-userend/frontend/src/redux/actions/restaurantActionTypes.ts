const restaurantActionTypes = {
  LOAD_CATEGORIES: 'LOAD_CATEGORIES',
  FILTER_CATEGORIES: 'FILTER_CATEGORIES',
  LOAD_RESTAURANTS: 'LOAD_RESTAURANTS'
};

export default restaurantActionTypes
;
