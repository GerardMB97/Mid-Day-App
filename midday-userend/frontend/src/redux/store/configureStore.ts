import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import rootReducer from '../reducers';
import initialState from './initialState';

function configureStore () {
  const composeEnhancers = compose;

  return createStore(
    rootReducer,
    initialState,
    composeEnhancers(applyMiddleware(reduxImmutableStateInvariant(), thunk))
  );
}

export default configureStore();
