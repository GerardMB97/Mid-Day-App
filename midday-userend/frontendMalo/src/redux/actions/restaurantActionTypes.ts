const restaurantActionTypes = {
  LOAD_CATEGORIES: 'LOAD_CATEGORIES'
};

export default restaurantActionTypes
;
