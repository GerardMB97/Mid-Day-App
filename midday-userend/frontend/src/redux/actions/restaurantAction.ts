import axios from 'axios';
import restaurantActionTypes from './restaurantActionTypes';

export const loadCategories = () => {
  return async (dispatch: Function) => {
    const categories = await axios.get('http://localhost:5000/api/midday/restaurants/categories');

    dispatch({
      type: restaurantActionTypes.LOAD_CATEGORIES,
      categories
    });
  };
}
;
