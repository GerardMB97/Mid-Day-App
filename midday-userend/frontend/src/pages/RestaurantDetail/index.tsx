import React, { useEffect } from 'react';
import { Dispatch, AnyAction, bindActionCreators } from 'redux';
import { Text, ImageBackground, StyleSheet, View, TouchableWithoutFeedback, TouchableOpacity } from 'react-native';

import { State } from '../../models';
import { getSelectedRestaurant } from '../../redux/actions/restaurantAction';
import { connect } from 'react-redux';
import colors from '../../../colors';
import { Calendar, CalendarList, Agenda } from 'react-native-calendars';
import Icon from 'react-native-vector-icons/Ionicons';
import TimePicker from '../../components/TimePicker';
import { checkSelectedHour, handleConfirm } from '../../utils';
import availableHours from '../../constants/availableHours';

const styles = StyleSheet.create({
  container: {
    width: '90%',
    flex: 1,
    alignSelf: 'center',
    alignItems: 'center',
    paddingTop: 20,
    position: 'relative'
  },
  imageContainer: {
    width: '100%',
    height: 200,
    borderRadius: 15,
    position: 'relative',
    borderColor: 'black',
    borderWidth: 1,
    overflow: 'hidden'
  },
  image: {
    height: '100%',
    width: '100%',
    borderRadius: 15,
    position: 'relative'
  },
  topContainer: {
    width: '100%',
    height: 40,
    paddingLeft: 15,
    paddingRight: 15,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    backgroundColor: colors.transparentGray
  },
  bottomContainer: {
    width: '100%',
    height: 40,
    paddingLeft: 15,
    paddingRight: 15,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    backgroundColor: colors.transparentGray,
    position: 'absolute',
    top: 160
  },
  day: {
    marginTop: 20,
    width: '100%',
    height: 70,
    borderRadius: 15,
    padding: 10,
    alignItems: 'center'
  },
  calendar: {
    position: 'absolute',
    left: 70,
    top: 320
  },
  touchable: {
    width: '100%',
    height: '100%',
    position: 'absolute',
    zIndex: 0
  },
  bigFont: {
    fontSize: 18,
    marginBottom: 10
  },
  calendarIconContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    position: 'relative'
  },
  calendarIcon: {
    fontSize: 30,
    position: 'absolute',
    left: 180,
    top: -35
  },
  bookingDetails: {
    marginTop: 40,
    height: 100,
    width: '100%',
    borderColor: colors.green,
    borderWidth: 1,
    borderRadius: 15,
    position: 'absolute',
    top: 600
  },
  time: {
    fontSize: 15,
    width: '100%',
    height: 30
  },
  timeContainer: {
    position: 'absolute',
    left: 220,
    top: -35
  },
  confirm: {
    marginTop: 10,
    width: 100,
    height: 40,
    backgroundColor: colors.green,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 15
  },
  confirmContainer: {
    position: 'relative',
    top: 400,
    left: 130

  },
  hourModal: {
    borderRadius: 15,
    height: 60,
    width: '100%',
    backgroundColor: colors.gray,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10,
    position: 'absolute',
    left: -265,
    zIndex: 5
  },
  modalText: {
    textAlign: 'center'
  }
});

function RestaurantDetail ({ route, selectedRestaurant, actions }: any) {
  const [selectedDay, setSelectedDay] = React.useState(new Date().toLocaleDateString('es-ES').replace('/', '-').replace('/', '-'));
  const [calendarModal, setCalendarModal] = React.useState(false);
  const [selectedHour, setSelectedHour] = React.useState('12:00');
  const [wrongHourModal, setWrongHourModal] = React.useState(false);

  const handleConfirmCallback = () => {
    setWrongHourModal(false);
  };

  const { _id } = route.params;
  useEffect(() => {
    actions.getSelectedRestaurant(_id);
  }, []);

  return (

        <View style={styles.container}>
           <TouchableWithoutFeedback testID='calendarUnactive' style={styles.touchable} onPress={() => { setCalendarModal(false); }}><View style={styles.touchable}></View></TouchableWithoutFeedback>
       <View style = {styles.imageContainer} >

            <ImageBackground source= {{ uri: selectedRestaurant.image }} style = {styles.image} >

                <View style={styles.topContainer}>
                <Text>{selectedRestaurant.name}</Text>
                <Text>{selectedRestaurant.category.name}</Text>
                 </View>

                <View style={styles.bottomContainer}>
                  <Text>{`${selectedRestaurant.street}, ${selectedRestaurant.number}`}</Text>
                  <Text >{selectedRestaurant.menuprice} â‚¬</Text>
                </View>

            </ImageBackground>

          </View>
          <View style={styles.day}>
            <Text style={styles.bigFont}>Cuando quieres reservar?</Text>
            <View style = {styles.calendarIconContainer}>
              <View style={styles.timeContainer}>
            <TimePicker selectedHour={selectedHour} setSelectedHour= {setSelectedHour}/>
            </View>
            <Text>{selectedDay} a las {selectedHour}</Text>
            <TouchableWithoutFeedback testID='calendar' onPress={() => { setCalendarModal(true); }}><Icon name="calendar-outline" style={styles.calendarIcon}></Icon></TouchableWithoutFeedback>
            </View>
          </View>
          {calendarModal &&
          <View style = {styles.calendar}>
          <Calendar
          onDayPress={({ dateString }) => { setSelectedDay(dateString.split('-').reverse().join('-')); }}

          />
          </View>}
          <View style={styles.confirmContainer}>
            {wrongHourModal &&
            <View style={styles.hourModal}>
              <Text style={styles.modalText}>A las {selectedHour} estamos cerrados, nuestro horario es de 12:00 a 16:00</Text>
            </View>
            }

          <TouchableOpacity style={styles.confirm} onPress={() => handleConfirm(checkSelectedHour(selectedHour, availableHours), setWrongHourModal, handleConfirmCallback) }><Text>Reservar</Text></TouchableOpacity>
          </View>
          </View>
  );
}

function mapStateToProps ({
  restaurants: { selectedRestaurant }
}: {
  restaurants: State['restaurants'];
}) {
  return { selectedRestaurant };
}

function mapDispatchToProps (dispatch: Dispatch<AnyAction>) {
  return { actions: bindActionCreators({ getSelectedRestaurant }, dispatch) };
}

export default connect(mapStateToProps, mapDispatchToProps)(RestaurantDetail);
