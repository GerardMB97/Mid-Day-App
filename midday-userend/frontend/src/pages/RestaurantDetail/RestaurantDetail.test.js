import React from 'react';
import { fireEvent, render } from 'react-native-testing-library';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import * as actions from '../../redux/actions/restaurantAction';
import RestaurantDetail from '.';
import { getSelectedRestaurant } from '../../redux/actions/restaurantAction';

jest.mock('../../redux/actions/restaurantAction');
jest.mock('react-native-vector-icons/Ionicons', () => 'Icon');
jest.mock('react-native-calendars', () => 'Calendar');
jest.mock('react-native-modal-datetime-picker', () => 'TimePicker');

describe('Given a component Restaurant Detail', () => {
  const mockStore = configureStore();
  const store = mockStore({ restaurants: { selectedRestaurant: { category: {} } }, route: { params: { _id: 15 } } });

  jest.spyOn(actions, 'getSelectedRestaurant').mockReturnValue({ type: '' });
  test('Then it should render correctly', () => {
    const rendered = render(<Provider store={store}><RestaurantDetail route={{ params: 15 }}></RestaurantDetail></Provider>);

    expect(rendered).toMatchSnapshot();
  });
  describe('When buton with testId calendarUnactive is pressed', () => {
    test('Then it should render correctly', () => {
      const rendered = render(<Provider store={store}><RestaurantDetail route={{ params: 15 }}></RestaurantDetail></Provider>);
      const unactiveCalendar = rendered.getByTestId('calendarUnactive');
      fireEvent.press(unactiveCalendar);

      expect(rendered).toMatchSnapshot();
    });
  });
  describe('When buton with testId calendar is pressed', () => {
    test('Then it should render correctly', () => {
      const rendered = render(<Provider store={store}><RestaurantDetail route={{ params: 15 }}></RestaurantDetail></Provider>);
      const activeCalendar = rendered.getByTestId('calendar');
      fireEvent.press(activeCalendar);

      expect(rendered).toMatchSnapshot();
    });
  });
});
