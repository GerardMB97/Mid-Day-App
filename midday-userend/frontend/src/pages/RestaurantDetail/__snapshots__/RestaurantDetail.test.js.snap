// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a component Restaurant Detail Then it should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "paddingTop": 20,
      "position": "relative",
      "width": "90%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": "100%",
        "position": "absolute",
        "width": "100%",
        "zIndex": 0,
      }
    }
    testID="calendarUnactive"
  />
  <View
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 15,
        "borderWidth": 1,
        "height": 200,
        "overflow": "hidden",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "borderRadius": 15,
          "height": "100%",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(242, 242, 242, 0.8)",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "space-between",
            "paddingLeft": 15,
            "paddingRight": 15,
            "width": "100%",
          }
        }
      >
        <Text />
        <Text />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(242, 242, 242, 0.8)",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "space-between",
            "paddingLeft": 15,
            "paddingRight": 15,
            "position": "absolute",
            "top": 160,
            "width": "100%",
          }
        }
      >
        <Text>
          undefined, undefined
        </Text>
        <Text>
           €
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 15,
        "height": 70,
        "marginTop": 20,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "marginBottom": 10,
        }
      }
    >
      Cuando quieres reservar?
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "left": 220,
            "position": "absolute",
            "top": -35,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="show"
        >
          <Icon
            name="time-outline"
            style={
              Object {
                "fontSize": 30,
              }
            }
          />
          <TimePicker
            headerTextIOS="Estamos disponibles de 12:00 a 16:00 "
            isVisible={false}
            locale="en_GB"
            minuteInterval={30}
            mode="time"
            onCancel={[Function]}
            onConfirm={[Function]}
            testID="picker"
          />
        </View>
      </View>
      <Text>
        15-3-2021
         a las 
        12:00
      </Text>
      <Icon
        accessible={true}
        focusable={true}
        name="calendar-outline"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "fontSize": 30,
            "left": 180,
            "position": "absolute",
            "top": -35,
          }
        }
        testID="calendar"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "left": 130,
        "position": "relative",
        "top": 400,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(135, 175, 147)",
          "borderRadius": 15,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <Text>
        Reservar
      </Text>
    </View>
  </View>
</View>
`;

exports[`Given a component Restaurant Detail When buton with testId calendarUnactive is pressed Then it should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "paddingTop": 20,
      "position": "relative",
      "width": "90%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": "100%",
        "position": "absolute",
        "width": "100%",
        "zIndex": 0,
      }
    }
    testID="calendarUnactive"
  />
  <View
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 15,
        "borderWidth": 1,
        "height": 200,
        "overflow": "hidden",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "borderRadius": 15,
          "height": "100%",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(242, 242, 242, 0.8)",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "space-between",
            "paddingLeft": 15,
            "paddingRight": 15,
            "width": "100%",
          }
        }
      >
        <Text />
        <Text />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(242, 242, 242, 0.8)",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "space-between",
            "paddingLeft": 15,
            "paddingRight": 15,
            "position": "absolute",
            "top": 160,
            "width": "100%",
          }
        }
      >
        <Text>
          undefined, undefined
        </Text>
        <Text>
           €
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 15,
        "height": 70,
        "marginTop": 20,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "marginBottom": 10,
        }
      }
    >
      Cuando quieres reservar?
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "left": 220,
            "position": "absolute",
            "top": -35,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="show"
        >
          <Icon
            name="time-outline"
            style={
              Object {
                "fontSize": 30,
              }
            }
          />
          <TimePicker
            headerTextIOS="Estamos disponibles de 12:00 a 16:00 "
            isVisible={false}
            locale="en_GB"
            minuteInterval={30}
            mode="time"
            onCancel={[Function]}
            onConfirm={[Function]}
            testID="picker"
          />
        </View>
      </View>
      <Text>
        15-3-2021
         a las 
        12:00
      </Text>
      <Icon
        accessible={true}
        focusable={true}
        name="calendar-outline"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "fontSize": 30,
            "left": 180,
            "position": "absolute",
            "top": -35,
          }
        }
        testID="calendar"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "left": 130,
        "position": "relative",
        "top": 400,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(135, 175, 147)",
          "borderRadius": 15,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <Text>
        Reservar
      </Text>
    </View>
  </View>
</View>
`;
